### PR TITLE
New AzMarketplace Cmdlets

### PR DESCRIPTION
**Pull request by ilavi19**
## Description

This PR is adding CRUD operation for azure marketplace private store , such as adding collection to a store, adding marketplace offers to collection and so on

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
    *cant find this file
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        -https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1089
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
    where to set the markdown files? you can find them here https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1089

### CODE REVIEW

**Comment by wyunchi-ms** [comment]
Please remove the necessary files. You can use https://github.com/Azure/azure-powershell/tree/generation/src/Databricks as a guildline.

File: src/Marketplace/docs/Move-AzMarketplacePrivateStoreCollectionOffer.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+external help file:
+Module Name: Az.Marketplace
+online version: https://docs.microsoft.com/powershell/module/az.marketplace/move-azmarketplaceprivatestorecollectionoffer
+schema: 2.0.0
+---
+
+# Move-AzMarketplacePrivateStoreCollectionOffer
+
+## SYNOPSIS
+transferring offers (copy or move) from source collection to target collection(s)
```
Comment (wyunchi-ms) [inline src/Marketplace/docs/Move-AzMarketplacePrivateStoreCollectionOffer.md:11]: What's the difference with `Copy-AzMarketplacePrivateStoreCollectionOffer`? Is the operation always be `Move`?

File: src/Marketplace/docs/Move-AzMarketplacePrivateStoreCollectionOffer.md [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+```
+
+## DESCRIPTION
+transferring offers (copy or move) from source collection to target collection(s)
+
+## EXAMPLES
+
+### Example 1: Gets private store offers regardless of collections 
+```powershell
+PS C:\> Invoke-AzMarketplaceQueryPrivateStoreOffer -PrivateStoreId 3ac32d8c-e888-4dc6-b4ff-be4d755af13a
```
Comment (wyunchi-ms) [inline src/Marketplace/docs/Move-AzMarketplacePrivateStoreCollectionOffer.md:48]: The example is wrong.

File: src/Marketplace/docs/New-AzMarketplacePrivateStore.md [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+
+```
+New-AzMarketplacePrivateStore -Id <String> [-Availability <Availability>] [-Branding <Hashtable>]
+ [-ETag <String>] [-IsGov] [-NotificationSettingRecipient <IRecipient[]>]
+ [-NotificationSettingSendToAllMarketplaceAdmin] [-PrivateStoreName <String>] [-TenantId <String>]
+ [-DefaultProfile <PSObject>] [-PassThru] [-Confirm] [-WhatIf] [<CommonParameters>]
+```
+
+## DESCRIPTION
+Changes private store properties
```
Comment (wyunchi-ms) [inline src/Marketplace/docs/New-AzMarketplacePrivateStore.md:23]: Description cannot match the cmdlet name.

File: src/Marketplace/docs/Copy-AzMarketplacePrivateStoreCollectionOffer.md [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+
+## DESCRIPTION
+transferring offers (copy or move) from source collection to target collection(s)
+
+## EXAMPLES
+
+### Example 1: Moves or copy offers from source collection to target collections.
+```powershell
+
+```
```
Comment (wyunchi-ms) [inline src/Marketplace/docs/Copy-AzMarketplacePrivateStoreCollectionOffer.md:49]: Please remove this empty block.

File: src/Marketplace/Az.Marketplace.psd1 [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  CompatiblePSEditions = 'Core', 'Desktop'
+  Author = 'Microsoft Corporation'
+  CompanyName = 'Microsoft Corporation'
+  Copyright = 'Microsoft Corporation. All rights reserved.'
+  Description = 'Microsoft Azure PowerShell: Marketplace cmdlets'
+  PowerShellVersion = '5.1'
+  DotNetFrameworkVersion = '4.7.2'
+  RequiredAssemblies = './bin/Az.Marketplace.private.dll'
+  FormatsToProcess = './Az.Marketplace.format.ps1xml'
+  FunctionsToExport = 'Copy-AzMarketplacePrivateStoreCollectionOffer', 'Get-AzMarketplaceBillingPrivateStoreAccount', 'Get-AzMarketplaceCollectionToSubscriptionMapping', 'Get-AzMarketplacePrivateStore', 'Get-AzMarketplacePrivateStoreCollection', 'Get-AzMarketplacePrivateStoreCollectionOffer', 'Get-AzMarketplaceQueryPrivateStoreOffer', 'New-AzMarketplacePrivateStore', 'New-AzMarketplacePrivateStoreCollection', 'New-AzMarketplacePrivateStoreCollectionOffer', 'Remove-AzMarketplacePrivateStoreCollection', 'Remove-AzMarketplacePrivateStoreCollectionOffer', 'Set-AzMarketplaceBulkPrivateStoreCollectionAction', 'Set-AzMarketplacePrivateStore', 'Set-AzMarketplacePrivateStoreCollection', 'Set-AzMarketplacePrivateStoreCollectionOffer', '*'
```
Comment (wyunchi-ms) [inline src/Marketplace/Az.Marketplace.psd1:14]: Add `Get-AzMarketplaceCollectionPrivateStoreToSubscriptionMapping` here.

File: src/Marketplace/docs/Move-AzMarketplacePrivateStoreCollectionOffer.md (hunk 1 above)
Comment (ilavi19) [reply src/Marketplace/docs/Move-AzMarketplacePrivateStoreCollectionOffer.md:11]: only copy, fixed

File: src/Marketplace/Az.Marketplace.psd1 (hunk 5 above)
Comment (ilavi19) [reply src/Marketplace/Az.Marketplace.psd1:14]: its already there, the name of Cmdlet changed to "Get-AzMarketplaceCollectionToSubscriptionMapping"

File: src/Marketplace/examples/New-AzMarketplacePrivateStoreCollectionOffer.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
+### Example 1: Creates or updates offer to private store collection 
+```powershell
+PS C:\>$acc = @{Accessibility = "azure_managedservices_professional"}
+PS C:\> New-AzMarketplacePrivateStoreCollectionOffer -CollectionId fdb889a1-cf3e-49f0-95b8-2bb012fa01f1 -PrivateStoreId 7f5402e4-e8f4-46bd-9bd1-8d27866a606b  -OfferId aumatics.azure_managedservices -Plan $acc
+
+Name                           SystemDataCreatedAt SystemDataCreatedBy SystemDataCreatedByType SystemDataLastModifiedAt SystemDataLastModifiedBy SystemDataLastModifiedByType
+----                           ------------------- ------------------- ----------------------- ------------------------ ------------------------ ----------------------------
+aumatics.azure_managedservices
+
```
Comment (wyunchi-ms) [inline src/Marketplace/examples/New-AzMarketplacePrivateStoreCollectionOffer.md:9]: Add the end of code block.

**Comment by ilavi19** [comment]
<html>
<body>
<p style="-webkit-font-smoothing: antialiased; box-sizing: border-box; margin: 0px; color: rgb(36, 36, 36); font-family: &quot;Segoe UI&quot;, system-ui, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(232, 235, 250); text-decoration-style: initial; text-decoration-color: initial;">Hi , I have a problem with one of the commands, I got braking changes because the output type was different , the difference in the output type is because the previous version of the command was written with SDK and the current with autorest , do you have a fix for this issue, it is not suppose to be a breaking charge because it only a naming difference.
These are the breaking changes related :

The cmdlet 'Get-AzMarketplacePrivateStore' no longer has output type 'Microsoft.Azure.Commands.Marketplace.Models.PrivateStore.PSPrivateStore'.

The cmdlet 'Get-AzMarketplacePrivateStore' no longer supports the type 'Microsoft.Azure.Commands.Common.Authentication.Abstractions.Core.IAzureContextContainer' for parameter 'DefaultProfile'.

**Review by wyunchi-ms — APPROVED** [review]

**Pull request merged**